### PR TITLE
airbyte-ci: Update metadata steps to accomodate new docs 

### PR DESCRIPTION
**Pull request by bnchrch**
## Overview

This works off of `lmossman/upload-versioned-connector-docs` to add support for the new docs arguments

This PR also took some time to
1. Introduce a new primitive for simple ci command based steps. 
2. Removes standalone airbyet-ci commands for metadata upload and validate as they are not used anymore and dont fit the new paradigm

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py
Comment (lmossman) [inline airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:67]: @bnchrch what actually sets this `documentation_file_path` when this step is automatically run due to merging a change to a connector which bumps the version?

I expected to see some logic in this PR which constructs the path to the connector doc file but I'm not seeing that, so how does the auto-publish on merge do this?

Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:67]: Good question!

So we already have a Connector Class that holds onto paths for
1. connector code directory
2. docs
3. metadata
4. icon

You can see the docs path method here:
(https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/connector_ops/connector_ops/utils.py#L218)

The I mounted that path using that method here:
https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py#L79

and again when constructing  the command here:
https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py#L79

Comment (lmossman) [reply airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:67]: That makes it very clear, thanks for explaining!

**Comment by lmossman** [comment]
@bnchrch I tried testing this out on this branch by running this command locally, with my env vars set to the `dev` buckets as you recommended [here](https://github.com/airbytehq/airbyte/pull/30410#issuecomment-1718435197):
```
airbyte-ci connectors --name=source-jira publish --main-release
```

And it gave me this error:
```
code/airbyte (bnchrch/ci/metadata-doc-step) ➜ airbyte-ci connectors --name=source-jira publish --main-release
[15:43:06] INFO     pipelines: Will run on the following 1 connectors: source-jira.                                                                                                                                                                                                                                                                                                  connectors.py:522
           INFO     pipelines: Running Dagger Command publish...                                                                                                                                                                                                                                                                                                                          utils.py:464
           INFO     pipelines: If you're running this command for the first time the Dagger engine image will be pulled, it can take a short minute...                                                                                                                                                                                                                                    utils.py:465
           INFO     pipelines: Saving dagger logs to: /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695706986/d5b09e2482883765cb79a1af09c6ab2542ec750c/dagger.log                                                                                                                   utils.py:475
Publishing from a local environment is not recommended and requires to be logged in Airbyte's DockerHub registry, do you want to continue? [y/N]: y
[15:43:10] WARNING  pipelines: Concurrency is forced to 1. For stability reasons we disable parallel publish pipelines.                                                                                                                                                                                                                                                              connectors.py:432
[15:43:14] INFO     pipelines: Dagger logs saved to /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695706986/d5b09e2482883765cb79a1af09c6ab2542ec750c/dagger.log                                                                                                                     utils.py:491
Traceback (most recent call last):
  File "/Users/lakemossman/.local/bin/airbyte-ci-internal", line 8, in <module>
    sys.exit(airbyte_ci())
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/sentry_utils.py", line 80, in wrapper
    return func(self, ctx, *args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/utils.py", line 482, in invoke
    pipeline_success = super().invoke(ctx)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/commands/groups/connectors.py", line 435, in publish
    publish_connector_contexts = anyio.run(
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_core/_eventloop.py", line 68, in run
    return asynclib.run(func, *args, **backend_options)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 204, in run
    return native_run(wrapper(), debug=debug)
  File "/Users/lakemossman/.pyenv/versions/3.10.8/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/lakemossman/.pyenv/versions/3.10.8/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 199, in wrapper
    return await func(*args)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/pipelines/connectors.py", line 101, in run_connectors_pipelines
    async with anyio.create_task_group() as tg_connectors:
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 597, in __aexit__
    raise exceptions[0]
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/publish.py", line 220, in run_connector_publish_pipeline
    metadata_upload_step = metadata.MetadataUpload(
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/pipelines/metadata.py", line 62, in __init__
    title = f"Upload metadata for {context.connector.technical_name} v{context.connector.docker_image_tag}"
AttributeError: 'ConnectorWithModifiedFiles' object has no attribute 'docker_image_tag'
```

I tried to make sure I ran the pipx install commands as instructed in the airbyte-ci/connectors/pipelines readme, but that doesn't seem to have helped

**Comment by bnchrch** [comment]
@lmossman Oh shoot! that was a really benign log change I made late thursday. It should be fixed now

**Comment by lmossman** [comment]
@bnchrch that fixed that specific issue, thanks!

I tried again and it successfully published source-jira to the dev bucket and uploaded the jira.md file as doc.md in the versioned and latest paths 👍 

However, there is one issue: it did not upload the `jira.inapp.md` file - only the full doc `.md` file was uploaded. Not sure why yet

**Comment by bnchrch** [comment]
@lmossman Ahh! yes `.inapp` that was because I did not add it to the CI step. Its added now.

**Comment by lmossman** [comment]
@bnchrch nice, so that fixed the case where a connector _does_ have an `.inapp.md` file.

But I tried running this on `desination-snowflake`, which doesn't have an inapp file, and it threw this error:
```
code/airbyte (bnchrch/ci/metadata-doc-step) ➜ airbyte-ci connectors --name=source-jira publish --main-release
[16:00:43] INFO     pipelines: Will run on the following 1 connectors: source-jira.                                                                                                                                                                                                                                                                                                  connectors.py:522
           INFO     pipelines: Running Dagger Command publish...                                                                                                                                                                                                                                                                                                                          utils.py:464
           INFO     pipelines: If you're running this command for the first time the Dagger engine image will be pulled, it can take a short minute...                                                                                                                                                                                                                                    utils.py:465
           INFO     pipelines: Saving dagger logs to: /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695708042/2cf0ea561ce36a59d1af836e2752d2d8c801ed88/dagger.log                                                                                                                   utils.py:475
Publishing from a local environment is not recommended and requires to be logged in Airbyte's DockerHub registry, do you want to continue? [y/N]: y
[16:00:51] WARNING  pipelines: Concurrency is forced to 1. For stability reasons we disable parallel publish pipelines.                                                                                                                                                                                                                                                              connectors.py:432
[16:00:54] INFO     Publish source-jira: Caching the latest CDK version...                                                                                                                                                                                                                                                                                                             contexts.py:242
[16:00:55] INFO     Publish source-jira - Validate metadata for source-jira: 🚀 Start Validate metadata for source-jira                                                                                                                                                                                                                                                                   bases.py:161
[16:01:07] INFO     Publish source-jira - Validate metadata for source-jira: ✅ was successful (duration: 12.02s)                                                                                                                                                                                                                                                                         bases.py:207
           INFO     Publish source-jira - Check if the connector docker image does not exist on the registry.: 🚀 Start Check if the connector docker image does not exist on the registry.                                                                                                                                                                                               bases.py:161
[16:01:09] INFO     Publish source-jira - Check if the connector docker image does not exist on the registry.: 🟡 was skipped (duration: 2.01s)                                                                                                                                                                                                                                           bases.py:205
           INFO     Publish source-jira: The connector version is already published. Let's upload metadata.yaml and spec to GCS even if no version bump happened.                                                                                                                                                                                                                       publish.py:254
           INFO     Publish source-jira - Upload connector spec to spec cache bucket: 🚀 Start Upload connector spec to spec cache bucket                                                                                                                                                                                                                                                 bases.py:161
[16:01:19] INFO     Publish source-jira - Upload connector spec to spec cache bucket: ✅ was successful (duration: 10.01s)                                                                                                                                                                                                                                                                bases.py:207
           INFO     Publish source-jira - Upload metadata for source-jira v0.6.0: 🚀 Start Upload metadata for source-jira v0.6.0                                                                                                                                                                                                                                                         bases.py:161
[16:01:33] INFO     Publish source-jira - Upload metadata for source-jira v0.6.0: 🟡 was skipped (duration: 14.02s)                                                                                                                                                                                                                                                                       bases.py:205
╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── SOURCE-JIRA - PUBLISH RESULTS ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                          Steps results                                                                                                                                                                                                                                                                                                                                             │
│ ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━┓                                                                                                                                                                                                                                                                                                    │
│ ┃ Step                                                                ┃ Result     ┃ Duration ┃                                                                                                                                                                                                                                                                                                    │
│ ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━┩                                                                                                                                                                                                                                                                                                    │
│ │ Validate metadata for source-jira                                   │ Successful │ 12.02s   │                                                                                                                                                                                                                                                                                                    │
│ │ Check if the connector docker image does not exist on the registry. │ Skipped    │ 2.01s    │                                                                                                                                                                                                                                                                                                    │
│ │ Upload connector spec to spec cache bucket                          │ Successful │ 10.01s   │                                                                                                                                                                                                                                                                                                    │
│ │ Upload metadata for source-jira v0.6.0                              │ Skipped    │ 14.02s   │                                                                                                                                                                                                                                                                                                    │
│ └─────────────────────────────────────────────────────────────────────┴────────────┴──────────┘                                                                                                                                                                                                                                                                                                    │
│ ℹ️  You can find more details with step executions logs in the saved HTML report.                                                                                                                                                                                                                                                                                                                   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ⏲️  Total pipeline duration for source-jira: 38.54s ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
           INFO     Publish source-jira: HTML report saved locally: /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695708042/2cf0ea561ce36a59d1af836e2752d2d8c801ed88/source-jira/0.6.0/output.html                                                                                  bases.py:626
           INFO     Publish source-jira: Opening HTML report in browser.                                                                                                                                                                                                                                                                                                                  bases.py:627
           INFO     Publish source-jira: Report saved locally at /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695708042/2cf0ea561ce36a59d1af836e2752d2d8c801ed88/source-jira/0.6.0/output.json                                                                                     bases.py:439
           INFO     Publish source-jira: Report saved locally at /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695708042/2cf0ea561ce36a59d1af836e2752d2d8c801ed88/complete.json                                                                                                     bases.py:439
           INFO     pipelines: Dagger logs saved to /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695708042/2cf0ea561ce36a59d1af836e2752d2d8c801ed88/dagger.log                                                                                                                     utils.py:491
code/airbyte (bnchrch/ci/metadata-doc-step) ➜ airbyte-ci connectors --name=destination-snowflake publish --main-release
[16:02:39] INFO     pipelines: Will run on the following 1 connectors: destination-snowflake.                                                                                                                                                                                                                                                                                        connectors.py:522
           INFO     pipelines: Running Dagger Command publish...                                                                                                                                                                                                                                                                                                                          utils.py:464
           INFO     pipelines: If you're running this command for the first time the Dagger engine image will be pulled, it can take a short minute...                                                                                                                                                                                                                                    utils.py:465
           INFO     pipelines: Saving dagger logs to: /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695708159/2cf0ea561ce36a59d1af836e2752d2d8c801ed88/dagger.log                                                                                                                   utils.py:475
Publishing from a local environment is not recommended and requires to be logged in Airbyte's DockerHub registry, do you want to continue? [y/N]: y
[16:02:42] WARNING  pipelines: Concurrency is forced to 1. For stability reasons we disable parallel publish pipelines.                                                                                                                                                                                                                                                              connectors.py:432
[16:02:46] INFO     pipelines: Dagger logs saved to /Users/lakemossman/code/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/publish/manual/bnchrch_ci_metadata-doc-step/1695708159/2cf0ea561ce36a59d1af836e2752d2d8c801ed88/dagger.log                                                                                                                     utils.py:491
Traceback (most recent call last):
  File "/Users/lakemossman/.local/bin/airbyte-ci-internal", line 8, in <module>
    sys.exit(airbyte_ci())
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/sentry_utils.py", line 80, in wrapper
    return func(self, ctx, *args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/utils.py", line 482, in invoke
    pipeline_success = super().invoke(ctx)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/commands/groups/connectors.py", line 435, in publish
    publish_connector_contexts = anyio.run(
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_core/_eventloop.py", line 68, in run
    return asynclib.run(func, *args, **backend_options)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 204, in run
    return native_run(wrapper(), debug=debug)
  File "/Users/lakemossman/.pyenv/versions/3.10.8/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/lakemossman/.pyenv/versions/3.10.8/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 199, in wrapper
    return await func(*args)
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/pipelines/connectors.py", line 101, in run_connectors_pipelines
    async with anyio.create_task_group() as tg_connectors:
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 597, in __aexit__
    raise exceptions[0]
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/publish.py", line 220, in run_connector_publish_pipeline
    metadata_upload_step = metadata.MetadataUpload(
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/pipelines/metadata.py", line 81, in __init__
    MountPath(context.connector.inapp_documentation_file_path, optional=True),
  File "<string>", line 5, in __init__
  File "/Users/lakemossman/.local/pipx/venvs/pipelines/lib/python3.10/site-packages/pipelines/steps/simple_ci_step.py", line 22, in __post_init__
    raise FileNotFoundError(message)
FileNotFoundError: docs/integrations/destinations/snowflake.inapp.md does not exist.
```

This should work for both cases, and if it doesn't have an inapp file, we should just not try to upload that

File: airbyte-ci/connectors/pipelines/pipelines/bases.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/bases.py:216]:
```suggestion
        raise NotImplementedError("Steps must define a '_run' attribute.")
```

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:46]: Suggestion: what about naming this `DockerRunStep`? Similar to `PoetryRunStep`, but I feel like it gives an indication of what will happen with the arguments

File: airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:96]: I'm a big fan of this abstraction 😍

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:46]: Although that might be a bit confusing since you don't pass in any info about what container to run (since it comes in through the context, right?)

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:55]:
```suggestion
            working_directory (str, optional): working directory to run the command in. Defaults to "/".
```

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:34]: Is there a good way to unit test this new type of step? 🙂 maybe something that runs some echoes of some env vars, mounted paths, etc? I'm reviewing the code but with less familiarity with dagger, test assertions would be helpful I feel

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:71]: Can probably leave this out until it's required - but I know a lot of volume mounting configuration usually allows to configure a separate destination path. if it's easy, could influence a different data type for mounts

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:110]: This means you can pass it explicitly or via context, right? Do we plan on later deprecating the latter?

File: airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:36]: ⚠️ I think we're missing the migration file path here!

But i'd definitely have noticed the spot to add it more quickly the first time if it were formatted like this!

Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:36]: migration file path needs to be in the list of mounted paths, but not in the documentation to upload (yet, at least) because it's used in the pre-upload validation, but never shown directly in the product. (Again, yet... I'm having thoughts about that modal...)

Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:36]: At least I'm pretty sure - though I can't find the line in the deleted code so maybe i'm thinking about something different

File: airbyte-ci/connectors/pipelines/pipelines/commands/groups/metadata.py
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/commands/groups/metadata.py:26]: ✌🏻 bye confusing double commands!

**Review by erohmensing — APPROVED** [review]
Looks good! Left way more comments than are blocking :D 
My wishes:
1. Please make sure that the migration file path stuff didn't get lost in the switchover - we need to have the migration file available during metadata validation in case there are breaking changes in the metadata. Not sure if it's missing or not. 
2. Would love a stab at unit tests for the new generic CI step since I think it will start to get used a lot. Knowing explicitly the behavior of mounts, env vars, etc could save us some future headache!

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:46]: Ah I like this! Should I keep `Simple`?

Im leaning towards yes, because Im not sure this should be the primary abstraction for all use cases. Just basic ones

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py:110]: Hmm I think more accurately its that you can either set the command at build or run time.

i.e. I can create a step using this class that can either
1. Only ever run one specific command 
2. Will run whatever command you give it at run time.

Can be useful if I want to change a steps arguments based on the output of a previous step

File: airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py
Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:36]: https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/connector_ops/connector_ops/qa_checks.py#L14

The check that the migration guide is properly formatted is part of QA checks. Not metadata
https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/connector_ops/connector_ops/qa_checks.py#L14

Though this is confusing. Theres definitely an argument to be made that this should be the same code location

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py]: > Im not sure this should be the primary abstraction for all use cases. Just basic ones

Makes sense to me - encourage people to build their own if things get complicated vs hacking more stuff into it. `SimpleDockerRunStep` seems good 👍🏻 Maybe a note in the docstring about "meant to serve as a base to handle simple use cases"

File: airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py]: ooooh. that makes sense! thanks. maybe `input_command` or something to differentiate between `self.command` would help here

File: airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py:31]: Aha. Was almost certain I was missing something... Agreed it is confusing, but I don't think that's a problem for this pr.

**Review by bnchrch — COMMENTED** [review]
@lmossman This should be good to go! I just verified test and publish with
* destination-snowflake
* source-jira

I'll approve your PR and we should be free
1. to merge this PR into yours 
2. then yours into master

Ill leave that in your hands as Im sure you'd like to do a test through of this on your end!

**Comment by lmossman** [comment]
@bnchrch I tested this on source-asana and destination-oracle and it worked as expected on both.

However, I tried running it on `source-postgres-strict-encrypt`, and it raised the error:
```
FileNotFoundError: docs/integrations/sources/postgres-strict-encrypt.md does not exist.
```

This means that publishing will fail for any strict-encrypt connectors, right? If so that needs to be fixed before merging

**Comment by bnchrch** [comment]
Nooo. Strict encrypt variants!

Reading your comment I think youre on the right track. The only things I would add/modify is
1. to have a specific carve out if the connectors documentationUrl is not owned by airbyte
2. to update the docsPath argument to point to the top level docs

That means we should change the following
1. Use the documentationUrl as the source of truth
1. If its not an airbyte docs url then skip docs check / upload (maybe we can query for a valid status code?)
1. If it is an airbyte docs url then use that urls structure to get our docs path structure
1. Update the documentationPath argument we pass to the metadata_service ci to be the path to the top level of the docs folder (e.g. `docs/` instead of `docs/integrations/`

Thoughts?

**Review by lmossman — APPROVED** [review]
LGTM and works well when testing locally. Going to merge into my branch!

**Pull request merged**
